src/Nethermind/Nethermind.Abi.Test/Json/AbiParameterConverterTests.cs

```diff
@@ -0,0 +1,93 @@
+                yield return new TestCaseData("uint256", new AbiUInt(256), null);
+                yield return new TestCaseData("address", AbiType.Address, null);
+                yield return new TestCaseData("bool", AbiType.Bool, null);
+                yield return new TestCaseData("fixed", AbiType.Fixed, null);
+                yield return new TestCaseData("Ufixed", AbiType.UFixed, null);
+                yield return new TestCaseData("fixed16x10", new AbiFixed(16, 10), null);
+                yield return new TestCaseData("ufixed256x80", new AbiUFixed(256, 80), null);
+                yield return new TestCaseData("fixed96x1", new AbiFixed(96, 1), null);
+                yield return new TestCaseData("bytes", AbiType.DynamicBytes, null);
+                yield return new TestCaseData("bytes32", new AbiBytes(32), null);
+                yield return new TestCaseData("function", AbiType.Function, null);
+                yield return new TestCaseData("string", AbiType.String, null);
+                yield return new TestCaseData("int[]", new AbiArray(AbiType.Int256), null);
+                yield return new TestCaseData("string[5]", new AbiFixedLengthArray(AbiType.String, 5), null);
+                
+                yield return new TestCaseData("tuple", null, new NotSupportedException());
+                
+                yield return new TestCaseData("int1", null, new ArgumentException());
+                yield return new TestCaseData("int9", null, new ArgumentException());
+                yield return new TestCaseData("int300", null, new ArgumentException());
+                yield return new TestCaseData("int3000", null, new ArgumentException());
+                yield return new TestCaseData("fixed80", null, new ArgumentException());
+                yield return new TestCaseData("fixed80x81", null, new ArgumentException());
+                yield return new TestCaseData("bytes33", null, new ArgumentException());
+            }
+        }
+
+        [TestCaseSource(nameof(TypeTestCases))]
+        public void Can_read_json(string type, AbiType expectedType, Exception expectedException = null)
+        {
+            var converter = new AbiParameterConverter();
+            var model = new {name = "theName", type};
+            string json = JsonConvert.SerializeObject(model);
+            using (var jsonReader = new JsonTextReader(new StringReader(json)))
+            {
+                try
+                {
+                    var result = converter.ReadJson(jsonReader, typeof(AbiParameter), null, false, new JsonSerializer());
+                    var expectation = new AbiParameter() {Name = "theName", Type = expectedType};
+                    expectedException.Should().BeNull();
+                    result.Should().BeEquivalentTo(expectation);
+                }
+                catch (Exception e)
+                {
+                    if (e.GetType() != expectedException?.GetType())
```

Comment:
Assert.Throws<>

Reply:
won't work, we don't have a generic type here, can't make it work otherwise.